internal/controller/helmchart_controller.go

```diff
@@ -521,6 +522,10 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, e.Err.Error())
 		return sreconcile.ResultEmpty, e
 	}
+	if certsTmpDir != "" {
+		// this happens when the tls certs are stored in a temporary directory
+		defer os.RemoveAll(certsTmpDir)
+	}
```

Comment:
```suggestion
	defer os.RemoveAll(certsTmpDir)
```

should be sufficient, `os.RemoveAll` simply returns early on an empty string. See:

- https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/os/removeall_at.go;l=15-20
- https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/os/removeall_noat.go;l=16-20